modules/secrets-store-aws-secrets-manager.adoc

```diff
@@ -0,0 +1,245 @@
+              mountPropagation: HostToContainer
+      tolerations:
+      - operator: Exists
+      volumes:
+        - name: providervol
+          hostPath:
+            path: "/etc/kubernetes/secrets-store-csi-providers"
+        - name: mountpoint-dir
+          hostPath:
+            path: /var/lib/kubelet/pods
+            type: DirectoryOrCreate
+      nodeSelector:
+        kubernetes.io/os: linux
+----
++
+[NOTE]
+====
+This configuration is modified from the configuration provided in the link:https://github.com/aws/secrets-store-csi-driver-provider-aws#installing-the-aws-provider[AWS documentation] so that it works properly with {product-title}.
+====
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f aws-provider.yaml
+----
+
+. Create a secret provider class to define your secrets store provider:
+
+.. Create a YAML file that defines the `SecretProviderClass` object:
++
+.Example `secret-provider-class-aws.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-aws-provider                   <1>
+  namespace: my-namespace                 <2>
+spec:
+  provider: aws                           <3>
+  parameters:                             <4>
+    objects: |
+      array:
+        - |
+          objectName: tls-cert
+          objectType: secret
+----
+<1> Specify the name for the provider secret provider class.
+<2> Specify the namespace for the provider secret provider class.
+<3> Specify the provider as `aws`.
+<4> Specify the provider-specific configuration parameters.
+
+.. Create the `SecretProviderClass` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret-provider-class-aws.yaml
+----
+
+. Create a deployment to use this secret provider class:
+
+.. Create a YAML file that defines the `Deployment` object:
++
+.Example `deployment.yaml`
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-aws-deployment                              <1>
+  namespace: my-namespace                              <2>
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: my-storage
+  template:
+    metadata:
+      labels:
+        app: my-storage
+    spec:
+      containers:
+      - name: busybox
+        image: k8s.gcr.io/e2e-test-images/busybox:1.29
+        command:
+          - "/bin/sleep"
+          - "10000"
+        volumeMounts:
+        - name: secrets-store-inline
+          mountPath: "/mnt/key-store"
+          readOnly: true
+      volumes:
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: "my-aws-provider" <3>
+----
+<1> Specify the name for the deployment.
+<2> Specify the namespace deployment.
+// TODO: Does this have to be the same ns as the secret provider class?
```

Comment:
@xingxingxia @ropatil010 I used the same namespace for the deployment as the secret provider class object. Do they have to be in the same namespace, or could this deployment be in some other namespace?

Reply:
Yeah correct, it should be in the same namespace as per my understanding. 
cc: @xingxingxia

Reply:
Yes, https://secrets-store-csi-driver.sigs.k8s.io/concepts#secretproviderclass : "`NOTE: The SecretProviderClass needs to be created in the same namespace as the pod.`"

Reply:
Great I'll work that into the callout, thanks!